### PR TITLE
Add QA Framework, License File, and Clean up

### PR DESCRIPTION
**Pull request by nickpileggi947**
- Add support for a statistical quality framework on snowpipe process including template files and updated arguments.
- Remove any references from pipewrench in the comments/alerts to the user.
- Added a license file for Apache.

### CODE REVIEW

File: src/main/resources/templates/snowflake/shared/create-stage-qa-task.ssp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    call table_run_qa('${destination.stagingDatabase.schema}',
+                '${table.destinationName}',
+                ${destination.quality.minimumPercentage},
+                ${destination.quality.minimumCount},
+                ${destination.quality.minimumRuns},
+                ${destination.quality.standardDeviations});
+
+ALTER TASK ${destination.stagingDatabase.name}.${destination.stagingDatabase.schema}.${table.destinationName}_qa_task RESUME;
+
+ALTER TASK ${destination.stagingDatabase.name}.${destination.stagingDatabase.schema}.${table.destinationName}_qa_task SUSPEND;
```
Comment (brockn) [inline src/main/resources/templates/snowflake/shared/create-stage-qa-task.ssp:38]: How come we spend the the task at the end? Maybe a comment explaining this...

Comment (nickpileggi947) [reply src/main/resources/templates/snowflake/shared/create-stage-qa-task.ssp:38]: I could probably split those out into different templates.

**Comment by brockn** [comment]
I like this PR a lot. Feels like we should update the docs in customer-docs/ which will go here? https://docs.customer.phdata.io/docs/streamliner/latest/

Added Noor and Don as they have recently worked on Streamliner.

**Comment by nickpileggi947** [comment]
@brockn I do have an internal document I put together to describe the new framework. I could add that to the docs to explain how to use it?

**Comment by brockn** [comment]
@nickpileggi947 - yes that'd be great but since streamliner is available to customers to use directly we'll want that content in customer-docs. Should be easy since you have the content already.

**Comment by nickpileggi947** [comment]
@brockn Just pushed that into a change.

**Comment by brockn** [comment]
LGTM. Nice work!!

**Comment by brockn** [comment]
I'll give everyone another day to review and then merge.

**Comment by nickpileggi947** [comment]
@brockn Do we want to increment the version to 3.3 or leave it at 3.2?

**Comment by brockn** [comment]
Feels like 3.3 to me

File: customer-docs/user-manual.md [hunk 2]
```diff
@@ -379,6 +385,34 @@ The Snowpipe Append template is A Snowflake data pipeline to append newly arrivi
...
+
+To easily collect the data, we can add a stream to the table we are monitoring and have a task that runs the stored procedure to collect the data. Once the data is collected and stored, the stored procedure can check if the values are within the UCL and LCL or 3 standard deviations away from the historical mean. If there are any values outside the range, the stored procedure can call an external api alerting users that some of the date requires investigation. Below is a diagram showing the dataflow.
+
+![QA_process](../images/streamliner_qa-process.png)
+
+####Dos and Don'ts
+This process should be used when monitoring on ongoing pipeline. It will not tell you if something is wrong or confirm data fits a predefined structure.
+If will give fail a check if the amount of data spikes up or down, if fields
```
Comment (bpmcd) [inline customer-docs/user-manual.md:411]: Can you clarify the verbiage here? What does a failure look like or do? Does it send some alert or cause a process to exit uncleanly?

Comment (nickpileggi947) [reply customer-docs/user-manual.md:411]: @bpmcd I think the verbiage above would answer that question a bit more. I do agree that verbiage could be a bit clearer

> If there are any values outside the range, the stored procedure can call an external api alerting users that some of the date requires investigation.

**Comment by bpmcd** [comment]
Nice work @nickpileggi947 ! 

LGTM

**Review by bpmcd — APPROVED** [review]
LGTM

**Pull request merged**